docs: Create a documentation for GCP deployment and write addition informations about AWS (#1334)
aneoconsulting/ArmoniK: .docs/content/1.installation/4.gcp/0. gcp-all-in-one-deployment.md

Reviewer: It is good to had short information about the difference between the make destroy and the bootstrap destroy; specially the fact that the last one is not meant to be perform everytime

Author: I added an explanation. Is it clear and enough ?

Reviewer 2: spaces in file names

# GCP All-in-One Deployment Guide

This guide will help you deploy your ArmoniK project on Google Cloud Platform (GCP).

## Step 1: Preparation

### 1. Install Google CLI

When receiving your credentials (email and password).

> Download and install the Google CLI by following the instructions on the [following link](https://cloud.google.com/sdk/docs/install?hl=fr#deb)

Reviewer 3: this link will work only for ubuntu or eventually debian based linux distribution

Author: No, it will only put the debian based linux distribution as the default choice in the documentation.

### 2. Initial Setup

> After installation, authenticate using the provided credentials (email and password) and select the project you want to deploy the resources to.

Reviewer 3: ```suggestion
After installation, you can authenticate using your credentials (email and password) and select the project you want to deploy the resources on.
```

> Follow [following tutorial](https://cloud.google.com/docs/authentication/provide-credentials-adc?hl=fr#how-to) to authenticate into the CLI.

Reviewer 3: ```suggestion
Follow [following tutorial](https://cloud.google.com/docs/authentication/provide-credentials-adc?hl=fr#how-to) to authenticate using the CLI.
```

You can authenticate using the following command:

```bash
gcloud auth login
```

To configure the project, if you don't know the project ID, you can list all the projects using the following command:

```bash
gcloud projects list
```

To configure the project, use the following command:

```bash
gcloud config set project <PROJECT_ID>
```

Once authenticated, you should see a confirmation page with the message:

*You are now authenticated with the gcloud CLI!*

## Step 2: Deployment

Navigate to the [`**gcp folder**`](https://github.com/aneoconsulting/ArmoniK/blob/main/infrastructure/quick-deploy/gcp/) and follow the steps below to deploy your resources.

### 1. Bootstrap

Generate a prefix key ready in the GCP environment to deploy your resources.

```bash
make bootstrap-deploy PREFIX=<PREFIX_KEY>
```

### 2. Deploy

To deploy your resources, execute:

```bash
make deploy PREFIX=<PREFIX_KEY>
```

## Step 3: Cleanup

### 1. Destroy the deployment

> - Attention: After using the deployment, you have to make sure to destroy it to avoid any additional costs. The command will destroy all the resources created during the deployment and the project will be deleted. It will conserve the prefix key for future use and by that, the terraform state will be saved in the GCP environment.

Reviewer 3: ```suggestion
- Warning: After using the deployment, you have to make sure to destroy it to avoid any additional costs. The next command will destroy all the resources created during the deployment and the project will be deleted. It will only conserve the prefix key for future use. The terraform state will be saved in the GCP environment.
```

To destroy the deployment, use the following command:

```bash
make destroy PREFIX=<PREFIX_KEY>
```

### 2. Destroy the GCP Prefix Key - Optional

It's an optional step if you are not willing to use the prefix key in the future. The command will delete the prefix key and the terraform state from the GCP environment.

To clean up the GCP prefix key, use the following command:

```bash
make bootstrap-destroy PREFIX=<PREFIX_KEY>

> ```

Reviewer 2: This is not optional. It is not recommended at all. You should only do it when you are sure that you will not need it again. For instance, when you leave.

> It's an optional step if you are not willing to use the prefix key in the future. However, please note that if you delete the prefix key and the terraform state from the GCP environment, you will not be able to deploy on GCP again without it. It is not recommended to perform this step unless you are certain that you will not need the prefix key again, such as when you are leaving the project.

Reviewer 3: ```suggestion
It's an optional step if you are not willing to use the prefix key in the future. However, please note that if you delete the prefix key and the terraform state from the GCP environment, you will not be able to reproduce this exact deployment on GCP. It is not recommended to perform this step unless you are certain that you will not need the prefix key again, such as when you are leaving the project.
```

## Step 4: Add a Sample Partition

> Just like with an **AWS** or **localhost** deployment, you can add a sample partition to test deployment on the **GCP** environment. You need to build the images and redeploy the services after adding the sample partition in the parameters.tfvars file.

Reviewer 3: ```suggestion
Just as with an **AWS** or **localhost** deployment, you can add a sample partition to test a deployment on the **GCP** environment. You need to build the images and redeploy the services after adding the sample partition in the parameters.tfvars file.
```
